Palmer_McKay/index.py

```diff
@@ -0,0 +1,70 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from datetime import datetime, timezone
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from catalog import models as cmod
+import math
+
+@view_function
+def process_request(request, cat_id, pagenum:int=1):
+    # cat_id used to get Category, pagenum set to 1 by default if not given
+    try:
```

Comment:
You look pro using these try: statements!

Reply:
Sweet!